app/controllers/instructor-groups.js

```diff
@@ -18,59 +17,49 @@ export default Controller.extend({
   schoolId: null,
   titleFilter: null,
   showNewInstructorGroupForm: false,
-  newInstructorGroups: [],
-  instructorGroups: computed('selectedSchool', function(){
-    let defer = RSVP.defer();
+  newInstructorGroup: null,
+  deletedInstructorGroups: [],
+
+  instructorGroups: computed('selectedSchool', 'deletedInstructorGroups.[]', 'newInstructorGroup', async function(){
     let schoolId = this.get('selectedSchool').get('id');
-    if(isEmpty(schoolId)){
-      defer.resolve([]);
-    } else {
-      this.get('store').query('instructor-group', {
-        filters: {
-          school: schoolId
-        }
-      }).then(instructorGroups => {
-        defer.resolve(instructorGroups);
-      });
+    if(isEmpty(schoolId)) {
+      resolve([]);
     }
-
-    return PromiseArray.create({
-      promise: defer.promise
+    return await this.get('store').query('instructor-group', {
+      filters: {
+        school: schoolId
+      }
     });
   }),
 
   changeTitleFilter: task(function * (value) {
-    const clean = escapeRegExp(value);
-    this.set('titleFilter', clean);
+    this.set('titleFilter', value);
     yield timeout(250);
-
-    return clean;
+    return value;
   }).restartable(),
 
   hasMoreThanOneSchool: gt('model.schools.length', 1),
   filteredInstructorGroups: computed(
     'changeTitleFilter.lastSuccessful.value',
     'instructorGroups.[]',
-    function(){
-      let defer = RSVP.defer();
+    async function(){
       let title = this.get('changeTitleFilter.lastSuccessful.value');
-      let exp = new RegExp(title, 'gi');
-      this.get('instructorGroups').then(instructorGroups => {
-        let filteredInstructorGroups;
-        if(isEmpty(title)){
-          filteredInstructorGroups = instructorGroups;
-        } else {
-          filteredInstructorGroups = instructorGroups.filter(instructorGroup => {
-            return isPresent(instructorGroup.get('title')) && instructorGroup.get('title').match(exp);
-          });
-        }
-        defer.resolve(filteredInstructorGroups.sortBy('title'));
-      });
-
-
-      return PromiseArray.create({
-        promise: defer.promise
-      });
+      if (!isPresent(title)) {
```

Comment:
I don't understand why this double check? Can't we just depend on `this.get('titleFilter')` and not worry about the task value?

Reply:
you are absolutely correct. i'm simplifying this.